tests/pytests/unit/states/test_pkgrepo.py

```diff
@@ -0,0 +1,53 @@
+"""
+    :codeauthor: Tyler Johnson <tjohnson@saltstack.com>
+"""
+
+import pytest
+import salt.states.pkgrepo as pkgrepo
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {
+        pkgrepo: {
+            "__opts__": {"test": True},
+            "__grains__": {"os": "", "os_family": ""},
+        }
+    }
+
+
+def test_new_key_url():
+    """
+    Test when only the key_url is changed that a change is triggered
+    """
+    kwargs = {
+        "name": "deb http://mock/ sid main",
+        "disabled": False,
+    }
+    key_url = "http://mock/changed_gpg.key"
+
+    with patch.dict(pkgrepo.__salt__, {"pkg.get_repo": MagicMock(return_value=kwargs)}):
+        ret = pkgrepo.managed(key_url=key_url, **kwargs)
+        assert {"key_url": {"old": None, "new": key_url}} == ret["changes"]
```

Comment:
In pytesr we assert what we got against what we expext. In case of failure and if the object can be `diff`'ed, the diff will make sense...
```suggestion
        assert ret["changes"] == {"key_url": {"old": None, "new": key_url}}
```